ex1/report/chapters/development.tex

```diff
@@ -0,0 +1,62 @@
+\documentclass[../main.tex]{subfiles}
+
+\section{Development}
+The development was carried out in natural stages of adding complexity.
+
+\subsection{Lighting the LEDs}
+The first course of action was to allow the program to control the LEDs.
+This allows for some rudimentary output, and is a nessesary first step to complete the assignment.
+
+The EFM32GG has external controllers memory mapped.
+One of them is a GPIO-controller.
+At startup, to save power, the external controllers are not clocked.
+As such it is not possible to comunicate with the controller.
+Thus before the GPIO-controller can be configured, the CMU must be set to enable the HFPERCLK for the GPIO-controller.
+This is done by setting bit 13 in the CMU\_HFPERCLKEN0 register.
+
+\subsection{Enabling and using the gamepad}
+
+After the setup for the GPIO output was finished we enabled the input from the gamepad. The first thing was to enable the pull up resisters on the GPIO pins C0 to C7 by writing 0xff to GPIO\_DOUT.  
+
+\begin{lstlisting}[frame=single]
+mov r2, 0xff
+str r2, [r1, GPIO_DOUT]
+\end{lstlisting}
+
+Then the gamepad buttons was set as inputs with the pull and glitch-filter by writing 0x33333333 to GPIO\_PC\_MODEL. 
+
+
+\begin{lstlisting}[frame=single]
+ldr r2, =0x33333333
+str r2, [r1, GPIO_MODEL]
+\end{lstlisting}
+
+\subsection{Interrupts}
+	\label{ss:interrupts}
+\subsection{Demonstration}
+
+As an extra learning exercise for the team, a small demo was implemented. After the interrupt was triggered and the code in gpio\_handler was executed when a button was pushed as described in the section \ref{ss:interrupts}. 
```

Comment:
Please use one line per sentence. It's hard to work with this in git, and I can't comment on specific sentences here on GitHub.